account_analytic_distribution_manual/hooks.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import SUPERUSER_ID, api, tools
+
+
+def post_init_hook(cr, registry):
+    if tools.table_exists(cr, "account_analytic_tag"):
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        sql = """
+            SELECT tag.name, tag.active, tag.company_id,
+                distribution.account_id, distribution.percentage
+            FROM account_analytic_distribution distribution
+                INNER JOIN account_analytic_tag tag ON tag.id = distribution.tag_id
+            WHERE active_analytic_distribution = true
+        """
+        env.cr.execute(sql)
+        distribution_by_tag = {}
+        distribution_manual_vals = []
+        for data in env.cr.dictfetchall():
+            tag_key = (data["name"], data["active"], data["company_id"])
```

Comment:
In my tests, I have found that there are duplicated tag names between active and no active, provoking an unique name, but maybe the problem is wider, as it seems previous tags didn't have an unique name constraint, so this code is going to mix the contents of all the tags with the same name.

Reply:
I updated the SQL. Please let me know if you have any comments.

With these tags:
![image](https://github.com/user-attachments/assets/8dc00b97-6e82-4383-a955-177bc3785888)

The result obtained is:
![image](https://github.com/user-attachments/assets/175c9ed2-9cc1-422d-84e8-b065a0782ce7)